recipes-sota/aktualizr/aktualizr-shared-prov-creds.bb

```diff
@@ -9,18 +9,48 @@ inherit allarch
 DEPENDS = "aktualizr-native zip-native"
 ALLOW_EMPTY_${PN} = "1"
 
+T = "${WORKDIR}/tmp"
+
 require credentials.inc
 
 do_install() {
     if [ -n "${SOTA_PACKED_CREDENTIALS}" ]; then
         install -m 0700 -d ${D}${localstatedir}/sota
-        cp "${SOTA_PACKED_CREDENTIALS}" ${D}${localstatedir}/sota/sota_provisioning_credentials.zip
-        # Device should not be able to push data to treehub
-        zip -d ${D}${localstatedir}/sota/sota_provisioning_credentials.zip treehub.json
-        # Device has no use for the API Gateway. Remove if present. See:
-        # https://github.com/advancedtelematic/ota-plus-server/pull/1913/
-        if unzip -l ${D}${localstatedir}/sota/sota_provisioning_credentials.zip api_gateway.url > /dev/null; then
-            zip -d ${D}${localstatedir}/sota/sota_provisioning_credentials.zip api_gateway.url 
+        # root.json contains the root metadata for bootstrapping the Uptane metadata verification process.
```

Comment:
Since it's not using a loop as some missing elements cause warnings and others not, we might as well group the two different policies together: first the ones with warning, then the others.

Reply:
Actually, some of these shouldn't be re-zipped at all. The various targets files and tufrepo.url should be left out here; they aren't used by libaktualizr. Then the three remaining items (root.json, autoprov.url, and the p12) should each have warnings, so maybe it is possible to get fancy with a loop.

Reply:
> The various targets files and tufrepo.url should be left out here

Do you mean this three files should not be zipped?

> remaining items (root.json, autoprov.url, and the p12) should each have warnings, so maybe it is possible to get fancy with a loop.

With an infinite loop?

Reply:
> Do you mean this three files should not be zipped?

Yes, libaktualizr has no use for them.

> With an infinite loop?

No, you can make the unzip and rezip logic generic by looping over the list of strings "root.json", "autoprov.url", and "autoprov_credentials.p12". I think that's what @lbonn was referring to when he mentioned a loop; please correct me if I misunderstood!

Reply:
yes, that's what I meant

Reply:
Oh, I see !